sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -226,3 +221,30 @@ export class AzurePowerShellCredential implements TokenCredential {
     });
   }
 }
+
+/**
+ *
+ * @internal
+ */
+export async function parseJsonToken(
+  result: string,
+): Promise<{ Token: string; ExpiresOn: string }> {
+  const jsonRegex = /{[^{}]*}/g;
+  const matches = result.match(jsonRegex);
+  let resultWithoutToken = result;
+  if (matches) {
+    try {
+      for (const item of matches) {
+        const jsonContent = JSON.parse(item);
+        if (jsonContent?.Token) {
+          resultWithoutToken = resultWithoutToken.replace(item, "");
+          logger.getToken.warning(resultWithoutToken);
```

Comment:
nice, I like that we are logging the rest of the string

Reply:
Yess!